pkg/generate/sort_test.go

```diff
@@ -0,0 +1,160 @@
+package generate
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/openebs/api/v2/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/api/v2/pkg/apis/types"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestGenerate(t *testing.T) {
+	type args struct {
+		list v1alpha1.BlockDeviceList
+	}
+	tests := []struct {
+		name string
+		args args
+		want *DeviceList
+	}{
+		{"empty node LinkedList",
+			args{list: v1alpha1.BlockDeviceList{Items: []v1alpha1.BlockDevice{}}}, nil},
+		{"single node LinkedList",
+			args{list: v1alpha1.BlockDeviceList{Items: []v1alpha1.BlockDevice{goodBD1N1}}},
+			&DeviceList{goodBD1N1, nil},
+		},
+		{
+			"two node LinkedList",
+			args{list: v1alpha1.BlockDeviceList{Items: []v1alpha1.BlockDevice{goodBD1N1, goodBD1N2}}},
+			&DeviceList{goodBD1N1, &DeviceList{goodBD1N2, nil}},
+		},
+		{
+			"four node LinkedList",
+			args{list: v1alpha1.BlockDeviceList{Items: []v1alpha1.BlockDevice{goodBD1N1, goodBD1N2, goodBD1N3, goodBD2N1}}},
+			&DeviceList{goodBD1N1, &DeviceList{goodBD1N2, &DeviceList{goodBD1N3,
+				&DeviceList{goodBD2N1, nil}}}},
+		},
+		{
+			"five node LinkedList",
+			args{list: v1alpha1.BlockDeviceList{Items: []v1alpha1.BlockDevice{goodBD1N1, goodBD1N2, goodBD1N3, goodBD2N1, goodBD3N1}}},
+			&DeviceList{goodBD1N1, &DeviceList{goodBD1N2, &DeviceList{goodBD1N3,
+				&DeviceList{goodBD2N1, &DeviceList{goodBD3N1, nil}}}}},
+		},
```

Comment:
Yeah, I've written a helper method which takes a slice of ints which are basically the gigabytes(powers of 10) sizes of the BDs objects so that I don't need to write them all over again.